packages/lib/src/core/helpers/layoutHelper.js

```diff
@@ -35,6 +35,59 @@ export const calcTargetItemSize = (styles, smartCalc = false) => {
   }
 };
 
+export const mutateNumberOfImagesPerRow = (styles) => {
+  //This will be used in the masonry and grid presets
+  let res = {...styles}
+  if (
+    styles.scrollDirection === GALLERY_CONSTS.scrollDirection.VERTICAL || //relevant for grid, in Masonry its fixed to !oneRow
```

Comment:
I prefer to avoid mutation when possible
```
const res = {...styles}
if (styles.scrollDirection === GALLERY_CONSTS.scrollDirection.VERTICAL) {
    return Object.assign({}, res, ...)
)
return res;